### PR TITLE
added integration test and watch topo util functions

### PR DESCRIPTION
**Pull request by eesha-deepak**

### CODE REVIEW

**Comment by adibrastegarnia** [comment]
retest this please

File: test/p4rt/suite.go [hunk 1]
```diff
@@ -25,7 +25,7 @@ func (s *TestSuite) SetupTestSuite(c *input.Context) error {
 	umbrella := charts.CreateUmbrellaRelease()
 	r := umbrella.
 		Set("global.image.registry", registry).
-		Set("import.onos-cli.enabled", false). // not needed - can be enabled by adding '--set onos-umbrella.import.onos-cli.enabled=true' to helmit args for investigation
+		Set("import.onos-cli.enabled", true). // not needed - can be enabled by adding '--set onos-umbrella.import.onos-cli.enabled=true' to helmit args for investigation
```
Comment (adibrastegarnia) [inline test/p4rt/suite.go:28]: We don't need to enable this. You can enable via helmit command

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: test/p4rt/suite.go (hunk 1 above)
Comment (eesha-deepak) [reply test/p4rt/suite.go:28]: I left it there for ease during testing, but I can change it back now.

File: test/p4rt/testP4RTConnectivity.go [hunk 2]
```diff
@@ -5,19 +5,59 @@
...
+	assert.NoError(t, err)
+	assert.NotNil(t, client)
+
+	newSwitch, err := p4rtsimulator.NewSwitchEntity(name, version, serviceAddress, servicePort, deviceID, modelID, role)
+	assert.NoError(t, err)
+	err = client.Create(newSwitch)
+	assert.NoError(t, err)
+
+	client.WaitForTargetAvailable(ctx, t, topoapi.ID(targetID), 2*time.Minute)
 	defer p4rtsimulator.DeleteSimulator(t, simulator)
```
Comment (adibrastegarnia) [inline test/p4rt/testP4RTConnectivity.go:47]: You can call defer  immediately after creating the simulator

File: test/utils/p4rtsimulator/simulator.go [hunk 3]
```diff
@@ -7,18 +7,133 @@ package p4rtsimulator
...
+			},
+		},
+	}
+
+	if err := o.SetAspect(&topo.TLSOptions{Insecure: true, Plain: true}); err != nil {
+		return nil, err
+	}
+
+	timeout := defaultTimeout
+	if err := o.SetAspect(&topo.Configurable{
```
Comment (adibrastegarnia) [inline test/utils/p4rtsimulator/simulator.go:61]: We don't need configurable aspect. The aspects are based on what we use in yaml file

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: test/utils/p4rtsimulator/simulator.go [hunk 4]
```diff
@@ -7,18 +7,133 @@ package p4rtsimulator
...
+}
+
+// NewSwitchEntity creates a switch entity with specified information
+func NewSwitchEntity(name string, version string, serviceAddress string, servicePort string, deviceID uint64, modelID string, role string) (*topo.Object, error) {
+	o := topo.Object{
+		ID:   topo.ID(name),
+		Type: topo.Object_ENTITY,
+		Obj: &topo.Object_Entity{
+			Entity: &topo.Entity{
+				KindID: topo.ID("switch"),
```
Comment (adibrastegarnia) [inline test/utils/p4rtsimulator/simulator.go:81]: You should use predefined kind IDs from topo api https://github.com/onosproject/onos-api/blob/master/go/onos/topo/topo.go#L58

**Review by adibrastegarnia — CHANGES_REQUESTED** [review]

File: test/utils/p4rtsimulator/simulator.go [hunk 5]
```diff
@@ -7,18 +7,133 @@ package p4rtsimulator
...
 	"testing"
 	"time"
 )
 
 const (
 	// Maximum time for an entire test to complete
 	defaultTestTimeout = 3 * time.Minute
+
+	// SimulatorTargetType type for simulated target
+	SimulatorTargetType = "devicesim"
```
[the remark this answers is not in the document]
Comment (eesha-deepak) [reply test/utils/p4rtsimulator/simulator.go:26]: they are used in the functions below

**Comment by adibrastegarnia** [comment]
retest this please

**Comment by adibrastegarnia** [comment]
retest this please

**Review by adibrastegarnia — APPROVED** [review]

**Pull request merged**